Compiling the C part fails due to not importing unistd

## Expected behavior ##

<!-- A clear and concise description of what you want to do and what you think should happen. (Code to reproduce the behavior can be added below). -->

`python setup.py develop` successfully compiles MDAnalysis for me.

## Actual behavior ##

<!-- What happened instead. Add as much detail as you can. Include (copy and paste) stack traces and any output. -->

```
mdanalysis/package/MDAnalysis/analysis/encore/dimensionality_reduction/src/spe.c:197:22: error:
      implicit declaration of function 'getpid' is invalid in C99
      [-Werror,-Wimplicit-function-declaration]
    srand(time(NULL)+getpid()*getpid());
```

This is because `spe.c` does not import `unistd` because it doesn't think the Mac is `__unix__`:

https://github.com/MDAnalysis/mdanalysis/blob/b2b7bcbdf4e2fe4e221d5e67842c90cf317375e5/package/MDAnalysis/analysis/encore/dimensionality_reduction/src/spe.c#L30-L32

I'm probably missing something because I managed to compile MDAnalysis a month ago without any modifications. I first noticed this about a week ago?

## Code to reproduce the behavior ##

<!-- Show us how to reproduce the failure. If you can, use trajectory files from the test data. Use the code snipped below as a starting point. -->

``` bash
python setup.py develop
```

## Current version of MDAnalysis ##

- Which version are you using? (run `python -c "import MDAnalysis as mda; print(mda.__version__)"`) develop
- Which version of Python (`python -V`)? 3.7.6 and 3.8.5
- Which operating system? MacOS 10.15.5
- Compilers: clang 12.0.0

```console
$ gcc --version
Configured with: --prefix=/Library/Developer/CommandLineTools/usr --with-gxx-include-dir=/Library/Developer/CommandLineTools/SDKs/MacOSX10.15.sdk/usr/include/c++/4.2.1
Apple clang version 12.0.0 (clang-1200.0.26.2)
Target: x86_64-apple-darwin19.5.0
Thread model: posix
InstalledDir: /Library/Developer/CommandLineTools/usr/bin

$ clang --version
Apple clang version 12.0.0 (clang-1200.0.26.2)
Target: x86_64-apple-darwin19.5.0
Thread model: posix
InstalledDir: /Library/Developer/CommandLineTools/usr/bin
```

--- a/package/CHANGELOG
+++ b/package/CHANGELOG
@@ -53,7 +53,9 @@ Fixes
   * libmdaxdr and libdcd classes in their last frame can now be pickled
     (Issue #2878, PR #2911)
   * AtomGroup now are pickled/unpickled without looking for its anchored
-    Universe (PR #2893) 
+    Universe (PR #2893)
+  * ensure that unistd.h is included on macOS when compiling ENCORE's spe.c
+    (Issue #2934)
 
 Enhancements
   * Refactored analysis.helanal into analysis.helix_analysis

--- a/package/MDAnalysis/analysis/encore/dimensionality_reduction/src/spe.c
+++ b/package/MDAnalysis/analysis/encore/dimensionality_reduction/src/spe.c
@@ -27,12 +27,10 @@
 #include <time.h>
 #include <sys/types.h>
 
-#ifdef __unix__
-    #include <unistd.h>
-#endif
-
 #ifdef _WIN32
     #include <io.h>
+#else /* unix-like __unix__ || __APPLE__ */
+    #include <unistd.h>
 #endif
 
 #define EPSILON 1e-8

